### ISSUE TITLE
`Lint/Syntax` is able to be disabled by directive comments

### ISSUE THREAD

**Issue by Earlopain** [rubocop/rubocop#12663]
Consider this very simple example file:

```rb
foo*
```

This isn't valid syntax and RuboCop complains. If I change this snippet to the following no offense is detected anymore:

```rb
# rubocop:disable Lint/Syntax
foo*
```

My understanding is that `Lint/Syntax` should not be able to be disabled under any circumstance. https://docs.rubocop.org/rubocop/1.60/configuration.html
> The cop enabling process can be altered by setting DisabledByDefault or EnabledByDefault (but not both) to true. These settings override the default for all cops to disabled or enabled, except Lint/Syntax which is always enabled, regardless of the cops' default values (whether enabled, disabled or pending).

> There is one exception from the general rule above and that is Lint/Syntax, a special cop that checks for syntax errors before the other cops are invoked. It cannot be disabled and its severity (fatal) cannot be changed in configuration.

If this is considered a bug then `RedundantCopDisableDirective` and `RedundantCopEnableDirective` should probably detect this as an offense. (actually, probably not? It's not valid syntax after all, don't think that would work)

## RuboCop version

```
$ [bundle exec] rubocop -V
1.60.2 (using Parser 3.3.0.5, rubocop-ast 1.30.0, running on ruby 3.2.2) [x86_64-linux]
  - rubocop-performance 1.20.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.26.1
```

**Comment by pboling** [comment]
Test fixtures of Ruby code for many types of code (e.g. prism / ASTs) need to be able to be tested against invalid Ruby syntax in order to test certain failure behaviors.
Perhaps the solution would be to prevent the "todo" feature from automatically quieting the violations?

For future googlers - if you use `rubocop-gradual` the `autocorrect` tool is able to mark this failure as exempted in the `.rubocop_gradual.lock`.
